src/plugins/discover/public/application/use_headless.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { useLocation } from 'react-router-dom';
+import { encode } from '@kbn/rison';
+import { useDiscoverServices } from '../hooks/use_discover_services';
+
+// Pass these via config or?
+const timelineSearchParams = {
+  isOpen: 'true',
+  activeTab: 'discover',
+};
+
+export const useHeadlessRoutes = (config) => {
+  const { application } = useDiscoverServices();
+  const { headlessLocation } = config;
+  const location = useLocation();
+  if (headlessLocation) {
+    const { hash, search } = location;
+    const currentSearchParams = new URLSearchParams(search);
+    currentSearchParams.set('timeline', encode(timelineSearchParams));
+    const searchString = decodeURIComponent(currentSearchParams.toString());
+    const pathWithSearchAndHash = hash ? `?${searchString}#${hash}` : `?${searchString}`;
```

Comment:
This looks really neat. Only difference is `discover` uses hash and we do not.. Now I understand what you were saying regarding the use of mode ('embedded` or `standalone`) param. Unfortunately, that happen deep in the app and cannot be directly user